File: 牛客网：PAT真题练兵场-乙级/有几个PAT-暴力统计超时.c
/*
 @Date    : 2017-11-23 21:22:13
 @Author  : 酸饺子 (changzheng300@foxmail.com)
 @Link    : https://github.com/SourDumplings
 @Version : $Id$
*/

/*
https://www.nowcoder.com/pat/6/problem/4066
 */

#include <stdio.h>

#define MAX 100000

int main()
{
    char s[MAX+1];
    gets(s);
    int i, j, k;
    int count = 0;
    for (i = 0; s[i]; i++)
    {
        if (s[i] == 'P')
        {
            for (j = i + 1; s[j]; j++)
            {
                if (s[j] == 'A')
                {
                    for (k = j + 1; s[k]; k++)
                    {
                        if (s[k] == 'T')
                        {
                            count = (count + 1) % 1000000007;
                        }
                    }
                }
            }
        }
    }
    printf("%d\n", count);
    return 0;
}

File: 牛客网：PAT真题练兵场-乙级/有几个PAT-暴力统计超时.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# @Date    : 2017-11-23 21:14:20
# @Author  : 酸饺子 (changzheng300@foxmail.com)
# @Link    : https://github.com/SourDumplings
# @Version : $Id$

'''
https://www.nowcoder.com/pat/6/problem/4066
'''


def main():
    s = input()
    l = len(s)
    count = 0

    for p in range(l):
        if s[p] == 'P':
            for a in range(p + 1, l):
                if s[a] == 'A':
                    for t in range(a + 1, l):
                        if s[t] == 'T':
                            count += 1

    result = count % 1000000007
    print(result)


if __name__ == '__main__':
    main()
